patroni/ctl.py

```diff
@@ -554,17 +556,17 @@ def failover(config_file, cluster_name, master, candidate, force, dcs, scheduled
 
     config, dcs, cluster = ctl_load_config(cluster_name, config_file, dcs)
 
-    if cluster.leader is None:
+    if cluster.leader is None and not is_paused(cluster):
         raise PatroniCtlException('This cluster has no master')
 
-    if master is None:
+    if master is None and (not is_paused(cluster) or cluster.leader):
         if force:
             master = cluster.leader.member.name
         else:
             master = click.prompt('Master', type=str, default=cluster.leader.member.name)
 
-    if cluster.leader.member.name != master:
-        raise PatroniCtlException('Member {0} is not the leader of cluster {1}'.format(master, cluster_name))
+        if not is_paused(cluster) and cluster.leader.member.name != master:
```

Comment:
Something is wrong here.
 1. Why this block is moved to the right?
 2. It should check that: if the `master` variable is not None and the `cluster` has `leader` then the `leader.name` MUST match with the `master`. If it doesn't match exceptions must be raised.